etl/db.py

```diff
@@ -61,7 +61,7 @@ def open_db() -> Generator[DBUtils, None, None]:
             connection.close()
 
 
-def get_dataset_id(db_conn: MySQLdb.Connection, dataset_name: str) -> Any:
+def get_dataset_id(db_conn: Union[MySQLdb.Connection, None], dataset_name: str) -> Any:
```

Comment:
`Union[X, None]` is better written as `Optional[X]`.

It's also weird to pass in `None` as an argument, better to move the `db_conn` argument to the end and default it to `None`, so that you can just call `get_dataset_id(dataset_name)` and it will work.